crates/autopilot/src/domain/eth/mod.rs

```diff
@@ -33,6 +39,27 @@ pub struct TxId(pub H256);
 #[derive(Debug, Clone, Copy, PartialEq, Eq, PartialOrd, Ord, Hash, From, Into)]
 pub struct TokenAddress(pub H160);
 
+impl TokenAddress {
+    /// If the token is ETH, return WETH, thereby "wrapping" it.
+    pub fn wrap(self, weth: WethAddress) -> Self {
+        if self == ETH_TOKEN {
+            weth.into()
+        } else {
+            self
+        }
+    }
+}
+
+/// The address of the WETH contract.
+#[derive(Debug, Clone, Copy, From, Into)]
+pub struct WethAddress(pub TokenAddress);
```

Comment:
Shouldn't this rather be `NativeToken` since gnosis chain has `xdai` for example?
This would then also lend itself for a better comment:
`/// ERC20 representation of the chain's native token (e.g. WETH on mainnet).`

Reply:
XDAI is never put in here. We want to ensure that deployed weth contracts return WETH address, and that some functions (like wrapping function) work specifically with weth address for the purpose of wrapping ETH. It doesn't have any relation to XDAI or any other token, native or not.

Reply:
No, take this [order](https://explorer.cow.fi/gc/orders/0xb1d983809d7a5f10023796f396e220653f82999c658da6314bb9540716b3e49a014e32ce1e960afa039070290def171bd1d74d4f671696aa?tab=overview) for example. It has `buy_token: 0xeee...eee` and is buying `xDai` not `WETH`.